extensions/wistia/src/api.ts

```diff
@@ -0,0 +1,119 @@
+import fetch from "node-fetch";
+import { getPreferenceValues } from "@raycast/api";
+import { AccountInfo, EmbedObject, Preferences, WistiaApiError, WistiaMedia, WistiaProject } from "./types";
+
+export async function fetchEmbedCode({
+  accountUrl,
+  hashedId,
+}: {
+  accountUrl: string;
+  hashedId: string;
+}): Promise<EmbedObject> {
+  const preferences: Preferences = getPreferenceValues();
+  const username = "api";
+  const password = preferences.wistiaApiToken;
+  const url = `https://api.wistia.com/oembed?url=${accountUrl}/medias/${hashedId}&embedType=async`;
+  const authString = `${username}:${password}`;
+
+  const response = await fetch(url, {
+    method: "GET",
+    headers: {
+      Authorization: `Basic ${Buffer.from(authString).toString("base64")}`,
+    },
+  });
+
+  if (!response.ok) {
+    const errInfo = (await response.json()) as WistiaApiError;
+    throw errInfo;
+  }
+
+  return (await response.json()) as EmbedObject;
+}
+
+export async function fetchMedias(): Promise<WistiaMedia[]> {
+  const preferences: Preferences = getPreferenceValues();
+  const username = "api";
+  const password = preferences.wistiaApiToken;
+  const url = "https://api.wistia.com/v1/medias.json";
+  const authString = `${username}:${password}`;
+
+  const response = await fetch(url, {
+    method: "GET",
+    headers: {
+      Authorization: `Basic ${Buffer.from(authString).toString("base64")}`,
+    },
+  });
+
+  if (!response.ok) {
+    const errInfo = (await response.json()) as WistiaApiError;
+    throw errInfo;
+  }
+
+  return (await response.json()) as WistiaMedia[];
+}
+
+export async function fetchAccountInfo(): Promise<AccountInfo> {
+  const preferences: Preferences = getPreferenceValues();
+  const username = "api";
+  const password = preferences.wistiaApiToken;
+  const url = "https://api.wistia.com/v1/account.json";
+  const authString = `${username}:${password}`;
+  const response = await fetch(url, {
+    method: "GET",
+    headers: {
+      Authorization: `Basic ${Buffer.from(authString).toString("base64")}`,
+    },
+  });
+
+  if (!response.ok) {
+    const errInfo = (await response.json()) as WistiaApiError;
```

Comment:
Could be inline, since the variable is not in use, happens a few times in this file.